atelier-saulx/based
# ⌨️ VSCode Autocomplete Extension

### SUMMARY
Provides autocomplete suggestions with the names of Based.io Functions or Queries present in the project.

### DETAILS
Usage:
![VSCode](https://github.com/atelier-saulx/based/assets/16566338/3301da80-99c2-47d2-88c9-0f3e9e5fcea0)


### DEVELOPER IMPACT
Now is possible to list all the functions easily!

### TESTING
Is not possible to install it by the Markeplace yet, but you can test it manually untill then. 

On terminal:
* Clone the repo
* Go to `packages/vscode`
* Open the project using `code .`
* Start the extension in debug mode:
    * Press F5 in VSCode. This will open a new instance of VSCode with the extension loaded in debug mode.
    * Open a TypeScript or JavaScript project in the VSCode development instance.
    * Check for `based.config.ts` files in the project.
    * Use the `useBasedQuery` hook (only available in the `@based/solidjs` for now) in a TypeScript or JavaScript file. When you start typing the value of the first parameter of `useBasedQuery`, you'll see the suggestions coming from the extension.

### TODO
[Publish to VSCode Marketplace](https://code.visualstudio.com/api/working-with-extensions/publishing-extension).
We need to create an Organization in Azure DevOps (IDK if we already have that).

![Alt Text](https://i.giphy.com/media/v1.Y2lkPTc5MGI3NjExZ3ozdHNzNDVyM2Z4NDBnZnoyMGg0bXpxdHc4Z2V4eGZ2NnhieDRkMSZlcD12MV9pbnRlcm5hbF9naWZfYnlfaWQmY3Q9Zw/XIqCQx02E1U9W/giphy.gif)
